extide/o.apache.tools.ant.module/src-bridge/org/apache/tools/ant/module/bridge/impl/ForkedJavaOverride.java

```diff
@@ -162,219 +166,216 @@ public synchronized void setError(File[] error) {
 
         private class NbOutputStreamHandler implements ExecuteStreamHandler {
 
-            private Thread outTask;
-            private Thread errTask;
-            private Thread inTask;
-            private Copier outCopier, errCopier; // #212526
-            private FoldingHelper foldingHelper;
-
+            private final ExecutorService tasks;
+            private final FoldingHelper foldingHelper;
+            private Future inputTask;
+            private InputStream stdout, stderr;
+            private OutputStream stdin;
+            
             NbOutputStreamHandler() {
                 this.foldingHelper = new FoldingHelper();
+                this.tasks = Executors.newFixedThreadPool(3, (r) -> {
+                    Thread t = new Thread(Thread.currentThread().getThreadGroup(),
+                        r,
+                        "I/O Thread for " + getProject().getName()); // NOI18N
+                    t.setDaemon(true);
+                    return t;
+                });
             }
 
-            public void start() throws IOException {}
+            private void setCopier(InputStream inputStream, OutputStream os, boolean delegate, boolean err) {
+                if (os == null || delegate) {
+                    tasks.submit(new TransferCopier(inputStream, AntBridge.delegateOutputStream(err)));
+                } else {
+                    tasks.submit(new TransferCopier(inputStream, os));
+                }
+            }
+
+            public void start() throws IOException {
+                NbBuildLogger buildLogger = getProject().getBuildListeners().stream()
+                    .filter(o -> o instanceof NbBuildLogger)
+                    .map(o -> (NbBuildLogger)o)
+                    .findFirst()
+                    .orElse(null);
+                if (buildLogger != null) {
+                    tasks.submit(new MarkupCopier(stdout, Project.MSG_INFO, false, outEncoding, buildLogger, foldingHelper));
+                    tasks.submit(new MarkupCopier(stderr, Project.MSG_WARN, true, outEncoding, buildLogger, foldingHelper));
```

Comment:
second task should use `errEncoding`.

i think we could also store the encoding as Charset, this would remove lookups inside toString(... "encoding"). 

```diff
diff --git a/extide/o.apache.tools.ant.module/src-bridge/org/apache/tools/ant/module/bridge/impl/ForkedJavaOverride.java b/extide/o.apache.tools.ant.module/src-bridge/org/apache/tools/ant/module/bridge/impl/ForkedJavaOverride.java
index b9fbad1..4f91cca 100644
--- a/extide/o.apache.tools.ant.module/src-bridge/org/apache/tools/ant/module/bridge/impl/ForkedJavaOverride.java
+++ b/extide/o.apache.tools.ant.module/src-bridge/org/apache/tools/ant/module/bridge/impl/ForkedJavaOverride.java
@@ -25,6 +25,7 @@
 import java.io.InputStream;
 import java.io.OutputStream;
 import java.io.UnsupportedEncodingException;
+import java.nio.charset.Charset;
 import java.util.concurrent.ExecutorService;
 import java.util.concurrent.Executors;
 import java.util.concurrent.Future;
@@ -104,8 +105,8 @@
 
     private class NbRedirector extends Redirector {
 
-        private String outEncoding = System.getProperty("file.encoding"); // NOI18N
-        private String errEncoding = System.getProperty("file.encoding"); // NOI18N
+        private Charset outEncoding;
+        private Charset errEncoding;
 
         // #158492. In Ant 1.8.0 output redirection cannot be distinguished by OutputStream subclass type
         // (LogOutputStream vs OutputStreamFunneler$Funnel) as OutputStreamFunneler$Funnel is
@@ -115,6 +116,14 @@
 
         public NbRedirector(Task task) {
             super(task);
+            try {
+                Charset cs = Charset.forName(System.getProperty("file.encoding")); // NOI18N
+                outEncoding = cs;
+                errEncoding = cs;
+            } catch (IllegalArgumentException ex) {
+                outEncoding = Charset.defaultCharset();
+                errEncoding = Charset.defaultCharset();
+            }
         }
 
         public @Override ExecuteStreamHandler createHandler() throws BuildException {
@@ -123,12 +132,12 @@
         }
 
         public @Override synchronized void setOutputEncoding(String outputEncoding) {
-            outEncoding = outputEncoding;
+            outEncoding = Charset.forName(outputEncoding);
             super.setOutputEncoding(outputEncoding);
         }
 
         public @Override synchronized void setErrorEncoding(String errorEncoding) {
-            errEncoding = errorEncoding;
+            errEncoding = Charset.forName(errorEncoding);
             super.setErrorEncoding(errorEncoding);
         }
 
@@ -199,7 +208,7 @@
                     .orElse(null);
                 if (buildLogger != null) {
                     tasks.submit(new MarkupCopier(stdout, Project.MSG_INFO, false, outEncoding, buildLogger, foldingHelper));
-                    tasks.submit(new MarkupCopier(stderr, Project.MSG_WARN, true, outEncoding, buildLogger, foldingHelper));
+                    tasks.submit(new MarkupCopier(stderr, Project.MSG_WARN, true, errEncoding, buildLogger, foldingHelper));
                 } else {
                     setCopier(stdout, getOutputStream(), delegateOutputStream, false);
                     setCopier(stderr, getErrorStream(), delegateErrorStream, true);
@@ -274,7 +283,7 @@
 
         private final InputStream in;
         private final int logLevel;
-        private final String encoding;
+        private final Charset encoding;
         private final RequestProcessor.Task flusher;
         private final ByteArrayOutputStream currentLine;
         private final OutputWriter ow;
@@ -282,7 +291,7 @@
         private final AntSession session;
         private final FoldingHelper foldingHelper;
 
-        public MarkupCopier(InputStream in, int logLevel, boolean err, String encoding, NbBuildLogger buildLogger, FoldingHelper foldingHelper) {
+        public MarkupCopier(InputStream in, int logLevel, boolean err, Charset encoding, NbBuildLogger buildLogger, FoldingHelper foldingHelper) {
             this.in = in;
             this.logLevel = logLevel;
             this.err = err;

```

Reply:
Thanks for spotting the paste-o, i agree on the charset change, i just left it as is to avoid unnecessary changes and since that form is widely used everywhere else in java code and avoids the need to handle the exception.

Do you want me to apply this diff and re-push the pull request?

Reply:
feel free to apply the diff or implement it yourself, whatever you prefer. I simply refactored it to check if it would work and pasted my version.

You can add it to the commit and force push into the PR branch.

Reply:
btw the way i see this: the exception would happen anyway, just later in the toString(... "encoding") method. So, converting it early to `Charset` should be fine without breaking contracts.

Reply:
Well now this code has to make a decision on what to do - the only exception being handled (illegalargumentexception) happens only if the property is null, and that case would've previously thrown nullpointerexception in String:lookupCharset().  Either way having a fallback is potentially a behavioural change.

I looked at an alternative approach but then noticed ByteArrayOutputStream:toString(Charset) was introduced at Java 10, so it can't be used anyway can it?

For what it's worth I had looked into a different buffering mechanism previously that could just use String:new directly but it didn't seem worth the added code.

Reply:
> ByteArrayOutputStream:toString(Charset) was introduced at Java 10, so it can't be used anyway can it?

oh wow you are right. I haven't checked for that. Lets forget about this then - my mistake. Lets only fix the `errEncoding` and we are good.

just as sidenote:
all other exceptions the method throws are subtypes of IllegalArgumentException, so it is basically catching everything, not only the null case (but making this a multi-catch would be clearer probably).